src/Generators/Generators.jl

```diff
@@ -0,0 +1,17 @@
+module Generators
+
+using LinearAlgebra: rmul!
+using Random: AbstractRNG, GLOBAL_RNG
+using LightGraphs
+using LightGraphs: is_graphical
+
+abstract type GraphGenerator end
+
+include("staticgenerators.jl")
+include("randomgenerators.jl")
+include("smallgenerators.jl")
+include("euclideangenerators.jl")
+
+export GraphGenerator, RandomGenerator, SmallGraph, StaticGenerator
```

Comment:
This is where it gets fun. Generators are now structs that can be used by the new dispatch-based approach:
```
julia> using LightGraphs, LightGraphs.Generators, LightGraphs.SimpleGraphs
julia> gen = LightGraphs.Generators.Bull()
LightGraphs.Generators.Bull()
julia> SimpleGraph(gen)
{5, 5} undirected simple Int64 graph
julia> SimpleGraph{UInt8}(gen)
{5, 5} undirected simple UInt8 graph
```

Reply:
It is getting verbose here, maybe we should export some of the of the generators - if someone does not want to pollute their namespace, they can always use `import`

Reply:
Yes - we'll have this export discussion later. I was thinking originally that we export just the functions, but I think now that we should export the types as well.